src/Model/User.php

```diff
@@ -127,18 +127,18 @@ public static function getIdFromPasswordAuthentication($user_info, $password)
 	{
 		$user = self::getAuthenticationInfo($user_info);
 
-		if ($user['legacy_password']) {
-			if (password_verify(self::hashPasswordLegacy($password), $user['password'])) {
-				self::updatePassword($user['uid'], $password);
-
-				return $user['uid'];
-			}
-		} elseif (password_verify($password, $user['password'])) {
+		if (password_verify($password, $user['password'])) {
 			if (password_needs_rehash($user['password'], PASSWORD_DEFAULT)) {
 				self::updatePassword($user['uid'], $password);
 			}
 
 			return $user['uid'];
+		} elseif (!empty($user['legacy_password']) || strpos($user['password'], '$') === false) {
```

Comment:
In this configuration this could by an `elseif ([check legacy password])`, right?

Reply:
I can still better like to test for `$` first to avoid interfering with the `password_verify()` function too much https://github.com/friendica/friendica/pull/4782#issuecomment-379547061 but besides that yes, it should work